pkg/conversion/conversion_factory.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversion
+
+import (
+	"strings"
+
+	"github.com/kcp-dev/logicalcluster/v3"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/conversion"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	apisinformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/apis/v1alpha1"
+)
+
+// CRConverterFactory instantiates converters that are capable of converting custom resources between different API
+// versions. It supports CEL-based conversion rules from APIConversion resources and the "none" conversion strategy.
+type CRConverterFactory struct {
+	nopConverter     conversion.CRConverter
```

Comment:
Is this a costly thing to create ephemerally? Surprised we need a shared instance here

Reply:
No, it's not costly. Can create on the fly if you'd prefer.